src/compiler/crystal/macros/methods.cr

```diff
@@ -211,11 +211,46 @@ module Crystal
         run_args << @last.to_macro_id
       end
 
-      success, result = @program.macro_run(filename, run_args)
-      if success
-        @last = MacroId.new(result)
+      result = @program.macro_run(filename, run_args)
+      if result.status.success?
+        @last = MacroId.new(result.stdout)
       else
-        node.raise "Error executing run: #{original_filename} #{run_args.map(&.inspect).join " "}\n\nGot:\n\n#{result}\n"
+        command = "#{original_filename} #{run_args.map(&.inspect).join " "}"
+
+        message = IO::Memory.new
+        message << "Error executing run (exit code: #{result.status.exit_code}): #{command}\n"
+
+        if result.stdout.empty? && result.stderr.empty?
+          message << "\nGot no output."
+        else
+          message << "\e[0m" if Colorize.enabled?
```

Comment:
I actually think that `Colorize.reset` is a better API. Using `String#colorize` implies you want to do something to a string, and most importantly that whatever changes you made will be reset afterwards. With `String#colorize.reset` you don't get that, there's no way to restore the old colours at the end of the string, so I don't think that API makes sense.